### PR TITLE
deprecate -q parameter in CLI

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):** #178 

**Description:**

Deprecate -q parameter, as it's the only single-letter parameter other than -u (url) and can easily be confused with quiet

**PR Checklist:**

- [X] Code is formatted
- [X] Tests pass
- [X] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac-api-client/blob/main/CHANGELOG.md)

### CODE REVIEW

**Review by gadomski — COMMENTED** [review]
It looks like this PR is built on the branches `pv/aoi-example` and `pv/default-max-items` -- I'm guessing that was accidental?

**Comment by philvarner** [comment]
> It looks like this PR is built on the branches `pv/aoi-example` and `pv/default-max-items` -- I'm guessing that was accidental?

no, i didn't want to have to worry about merging -- I meant to make this a draft pr

File: pystac_client/cli.py [hunk 1]
```diff
@@ -120,29 +120,34 @@ def parse_args(args: List[str]) -> Dict[str, Any]:
...
+    )
     search_group.add_argument(
         "--filter",
         help="Filter on queryables using language specified in filter-lang parameter",
     )
     search_group.add_argument(
         "--filter-lang",
         help="Filter language used within the filter parameter",
-        default="cql-json",
+        default="cql2-json",
```
Comment (philvarner) [inline pystac_client/cli.py:144]: yet another place i missed renaming this

File: CHANGELOG.md [hunk 2]
```diff
@@ -38,6 +38,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   To a user, this is only visible as a large delay in the method call and/or the
   exhaustion of all available memory. The iterator methods `items()` or
   `item_collections()` should be used instead. [#206](https://github.com/stac-utils/pystac-client/pull/206)
+- CLI parameter `-q` is now deprecated and may be removed as early as v0.5.0. Use `--query` instead.
```
Comment (gadomski) [inline CHANGELOG.md:41]:
```suggestion
- CLI parameter `-q` is now deprecated and may be removed as early as v0.5.0. Use `--query` instead. [#215](https://github.com/stac-utils/pystac-client/pull/215)
```

File: pystac_client/cli.py [hunk 3]
```diff
@@ -120,29 +120,34 @@ def parse_args(args: List[str]) -> Dict[str, Any]:
...
     search_group.add_argument(
-        "-q",
         "--query",
         nargs="*",
         help="Query properties of form " "KEY=VALUE (<, >, <=, >=, = supported)",
     )
+    search_group.add_argument(
+        "-q",
+        nargs="*",
+        help="DEPRECATED. Query properties of form "
```
Comment (gadomski) [inline pystac_client/cli.py:134]:
```suggestion
        help="DEPRECATED. Use --query instead. Query properties of form "
```

File: pystac_client/cli.py [hunk 4]
```diff
@@ -193,6 +198,14 @@ def parse_args(args: List[str]) -> Dict[str, Any]:
         if "json" in parsed_args["filter_lang"]:
             parsed_args["filter"] = json.loads(parsed_args["filter"])
 
+    if "q" in parsed_args:
+        logger.warning("Argument -q is deprecated, use --query instead")
+        if "query" not in parsed_args:
+            parsed_args["query"] = parsed_args["q"]
+        else:
+            logger.error("Both -q and --query arguments specified, ignoring -q")
```
Comment (gadomski) [inline pystac_client/cli.py:206]: Hm, it seems strange to log an error but continue execution. Maybe `warning`?

**Review by gadomski — CHANGES_REQUESTED** [review]

File: pystac_client/cli.py (hunk 4 above)
Comment (philvarner) [reply pystac_client/cli.py:206]: I did error b/c I think it's unlikely that a user would intentionally mix these. Honestly, I think there's no chance that any user is ever going to encounter this, but if they do, I want to make it clear what the behavior is.

**Review by gadomski — APPROVED** [review]

**Pull request merged**